### ISSUE TITLE
Generate clients

### ISSUE THREAD

**Issue by moul** [gnolang/gno#620]
1. We can generate RPC clients that will implement Tendermint2 verbs + AddPKG, and Call.
2. We can generate either dynamically or statically per-contract clients. This tool can rely on the Internet to query the existing `vm/qfuncs` helper.

In conclusion, we can either use 1. and be done, or use 2. to call 1. without worrying about marshalling and enjoy a fully type-safe experience with code completion, etc.

cc @ilgooz, what's your opinion and expected future needs?

**Comment by tbruyelle** [comment]
About `cuelang`, unfortunately the code generation part hasn't evolved as expected since the last time I used it (a couple of years ago though). It's still a very good langage for data definition and validation, but it still lacks the ability to generate the code from it, it's still marked as *TODO* in [their documentation][0].

Because of that, it also doesn't expose a `service` keyword like in protobuf, to define the RPC interface.

In conclusion, I no longer think it's a good fit for our need...

[0]: https://github.com/cue-lang/cue#tooling
